apps/examples/araplayer/HLSParser.cpp

```diff
@@ -0,0 +1,281 @@
+
+			hls_play_list_t *mediaPlaylist = new hls_play_list_t;
+			mediaPlaylist->url = std::string(segUrl);
+			mediaPlaylist->type = play_list_type_t::MEDIA;
+			mediaPlaylist->bandwidth = bandwidth;
+			mediaPlaylist->parent = playlist;
+
+			playlist->vecUrl.push_back(mediaPlaylist->url);
+			playlist->mapMediaPlaylist.insert(make_pair(mediaPlaylist->url, mediaPlaylist));
+		} else if (strncmp(BEGIN_SEGMENT_URL, data, mBeginSegmentUrlLen) == 0) {
+			if (playlist->type == play_list_type_t::NONE) {
+				playlist->type = play_list_type_t::SINGLE;
+				if (playlist->parent == NULL) {
+					DEBUG_TRACE(LOG_DEBUG, "Single Media Playlist [%s]", url);
+					mMapMasterPlaylist.insert(make_pair(std::string(url), playlist));
+				}
+			}
+
+			data += mBeginSegmentUrlLen + 1;
+			int point = 0;
+			double duration = 0;
+			while (*data != ',') {
+				duration = duration * 10 + (*data - '0');
+				point *= 10;
+				if (*data == '.') {
+					point = 1;
+				}
+				++data;
+			}
+
+			if (point) {
+				duration /= point;
+			}
+
+			++data;
+			while (*data == '\r' || *data == '\n') {
+				++data;
+			}
+
+			idx = 0;
+			while (*data != '\r' && *data != '\n') {
+				segUrl[idx++] = *data++;
+			}
+
+			segUrl[idx] = 0;
+			playlist->vecUrl.push_back(std::string(segUrl));
+		} else {
+			++data;
+		}
+	}
+
+	playlist->parsingCnt++;
+	if (mCallbackObject != NULL && mCBGetPlaylistInfo != NULL) {
+		if (isFirstParsedMediaPlaylist(playlist)) {
+			pInfo.playlistUrl = playlist->parent->url;
+		} else {
+			pInfo.playlistUrl = url;
+		}
+
+		getMediaPlaylistInfo(playlist, pInfo);
+		mCBGetPlaylistInfo(mCallbackObject, pInfo.playlistUrl.c_str(), pInfo);
+	}
+
+	return 1;
+}
+
+int HLSParser::getPlaylistInfo(const char *url, int bandwidth, has_segment_info_t *outSegInfo)
+{
+	int ret = 0;
+	PlaylistIter iter = mMapMasterPlaylist.find(url);
+	if (iter == mMapMasterPlaylist.end() || iter->second->parsingCnt <= 0) {
+		DEBUG_TRACE(LOG_ERR, "there is no appropriate playlist with url: %s", url);
+		return -1;
+	}
+
+	std::map<std::string, hls_play_list_t *> &mapMediaPlaylist = iter->second->mapMediaPlaylist;
+	for (PlaylistIter iter = mapMediaPlaylist.begin(); iter != mapMediaPlaylist.end(); ++iter) {
+		hls_play_list_t *playlist = iter->second;
+		if (playlist->bandwidth <= bandwidth) {
+			outSegInfo->url = std::string(playlist->url);
+			outSegInfo->bandwidth = bandwidth;
+			ret = 1;
+		}
+	}
+
+	return ret;
+}
+
+int HLSParser::getSegmentInfo(const char *url, int idx, int bandwidth, has_segment_info_t *outSegInfo)
+{
+	PlaylistIter iter = mMapMasterPlaylist.find(url);
+	hls_play_list_t *playlist;
+
+	if (iter == mMapMasterPlaylist.end()) {
+		DEBUG_TRACE(LOG_ERR, "there is no appropriate playlist with url: %s", url);
+		return -1;
+	}
+
+	std::map<std::string, hls_play_list_t *> &mapMediaPlaylist = iter->second->mapMediaPlaylist;
+	if (mapMediaPlaylist.size() > 0) {
+		for (PlaylistIter iter = mapMediaPlaylist.begin(); iter != mapMediaPlaylist.end(); ++iter) {
```

Comment:
PlaylistIter already declared above, but there's no compile error??

Reply:
it is in bracket, no compilation error occurs.
But i need to fix it.